simde/x86/aes.h

```diff
@@ -0,0 +1,603 @@
+  0x17, 0x2b, 0x04, 0x7e, 0xba, 0x77, 0xd6, 0x26, 0xe1, 0x69, 0x14, 0x63, 0x55, 0x21, 0x0c, 0x7d};// f
+
+
+/*
+ * Transformation in the Cipher and Inverse Cipher in which a Round 
+ * Key is added to the State using an XOR operation. The length of a 
+ * Round Key equals the size of the State (i.e., for Nb = 4, the Round 
+ * Key length equals 128 bits/16 bytes).
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_add_round_key(uint8_t *state, uint8_t *w, uint8_t r) {
+
+  int Nb = softaes_Nb;
+  uint8_t c;
+
+  for (c = 0; c < Nb; c++) {
+    state[Nb*0+c] = state[Nb*0+c]^w[4*Nb*r+4*c+0];
+    state[Nb*1+c] = state[Nb*1+c]^w[4*Nb*r+4*c+1];
+    state[Nb*2+c] = state[Nb*2+c]^w[4*Nb*r+4*c+2];
+    state[Nb*3+c] = state[Nb*3+c]^w[4*Nb*r+4*c+3];
+  }
+}
+
+/*
+ * Transformation in the Cipher that takes all of the columns of the 
+ * State and mixes their data (independently of one another) to 
+ * produce new columns.
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_mix_columns(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t k[] = {0x02, 0x01, 0x01, 0x03}; // a(x) = {02} + {01}x + {01}x2 + {03}x3
+  uint8_t i, j, col[4], res[4];
+
+  for (j = 0; j < Nb; j++) {
+    for (i = 0; i < 4; i++) {
+      col[i] = state[Nb*i+j];
+    }
+
+    //coef_mult(k, col, res);
+    softaes_coef_mult_lookup(0, col, res);
+
+    for (i = 0; i < 4; i++) {
+      state[Nb*i+j] = res[i];
+    }
+  }
+}
+
+/*
+ * Transformation in the Inverse Cipher that is the inverse of 
+ * MixColumns().
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_inv_mix_columns(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t k[] = {0x0e, 0x09, 0x0d, 0x0b}; // a(x) = {0e} + {09}x + {0d}x2 + {0b}x3
+  uint8_t i, j, col[4], res[4];
+
+  for (j = 0; j < Nb; j++) {
+    for (i = 0; i < 4; i++) {
+      col[i] = state[Nb*i+j];
+    }
+
+    //coef_mult(k, col, res);
+    softaes_coef_mult_lookup(4, col, res);
+
+    for (i = 0; i < 4; i++) {
+      state[Nb*i+j] = res[i];
+    }
+  }
+}
+
+/*
+ * Transformation in the Cipher that processes the State by cyclically 
+ * shifting the last three rows of the State by different offsets. 
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_shift_rows(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t i, k, s, tmp;
+
+  for (i = 1; i < 4; i++) {
+    // shift(1,4)=1; shift(2,4)=2; shift(3,4)=3
+    // shift(r, 4) = r;
+    s = 0;
+    while (s < i) {
+      tmp = state[Nb*i+0];
+
+      for (k = 1; k < Nb; k++) {
+        state[Nb*i+k-1] = state[Nb*i+k];
+      }
+
+      state[Nb*i+Nb-1] = tmp;
+      s++;
+    }
+  }
+}
+
+/*
+ * Transformation in the Inverse Cipher that is the inverse of 
+ * ShiftRows().
+ */
+SIMDE_FUNCTION_ATTRIBUTES
+void softaes_inv_shift_rows(uint8_t *state) {
+
+  int Nb = softaes_Nb;
+  uint8_t i, k, s, tmp;
+
+  for (i = 1; i < 4; i++) {
+    s = 0;
+    while (s < i) {
+      tmp = state[Nb*i+Nb-1];
+
+      for (k = Nb-1; k > 0; k--) {
```

Comment:
[`error: implicit conversion loses integer precision: 'int' to 'uint8_t' (aka 'unsigned char')`](https://github.com/simd-everywhere/simde/actions/runs/6497247919/job/17645921814?pr=1072#step:8:1095)